### DISCUSSION TITLE
Bringing up okd failing on vmware

### DISCUSSION THREAD

**Discussion by robsee** [okd-project/okd#1410] [Q&A]
Hi All. I'm brand new to okd, and I'm trying to bring up the latest 4.11 version of okd on vmware. The bootstrap node eventually comes up, but the master node has the following: [systemd]
Failed Units: 2
  machine-config-daemon-pull.service
  node-valid-hostname.service

the node-valid-hostname seems to be related to dhcp not providing a hostname, which while annoying, I can work around. The machine-config-daemon-pull seems to be related to a problem with selinux and librt. Is there a way I can get around the 2nd issue, or am I doing something wrong ?

**Comment by nnachefski** [comment]
I have this exact same problem and had to roll back to 4.10 because of it.  It appears that the hostname is not being set properly.

If you use the latest 4.10 build it will work (4.10.0-0.okd-2022-07-09-073606).  Make sure you have a dhcpd server on the subnet as well.

**Reply by robsee** [reply]
You can "solve" the hostname issue by having your dhcp server also issue a hostname. For me that meant I had to do a static entry within dhcp. I think its the machine-config-daemon-pull that is causing me the bigger problem

**Reply by nnachefski** [reply]
but then you cannot do IPI and must use UPI deployment method.  You can do OKD 4.10 IPI w/ vsphere which has much less prep time for cluster deployments.

**Comment by vrutkovs** [comment]
Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)
